.github/workflows/compile.yml

```diff
@@ -2,6 +2,16 @@ name: CI
 
 on: [push, pull_request]
 
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
```

Comment:
These should probably be granted only to the deploy job, not the build job, so people who compromise the (incredibly complex, involving at least two Turing-complete languages) build step can't push stuff to pages.